testsuite/features/step_definitions/salt_steps.rb

```diff
@@ -271,14 +271,16 @@
   # Complicated code because the checkbox is not a <input type=checkbox> but an <i>
   xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-square-o']" if action == 'check'
   xpath_query = "//a[@id = '#{formula}']/i[@class = 'fa fa-lg fa-check-square-o']" if action == 'uncheck'
-  puts "Selected formulas [beginning]:\n#{find('#chooseFormulas')['innerHTML']}"
+  # WORKAROUND
+  # DOM refreshes content of chooseFormulas element by accessing it. Then conditions are evaluated properly.
+  _inner_element = find('#chooseFormulas')['innerHTML']
   if all(:xpath, xpath_query, wait: DEFAULT_TIMEOUT).any?
-    puts "Selected formulas [if]:\n#{find('#chooseFormulas')['innerHTML']}"
+    _inner_element = find('#chooseFormulas')['innerHTML']
```

Comment:
Could it be the first assignment (the one with the comment) is enough, and we do not need the other two?

After all, if the problem is to force access to DOM tree, once is enough, isn't it?

Reply:
we just need to access the element not the assignment, we coul just use:
`find('#chooseFormulas')['innerHTML']`

Reply:
That too, yes

Reply:
@calancha @Bischoff I know :-) It came to my mind too. My first intention was to keep this solution as close as possible to the previous one. But it is also true, that the other two additional puts were present only for debug print. So... I will change it ;-)

Reply:
@lkotek maybe give it an extra testing shot with those changes, just in case?